pytidb/search.py

```diff
@@ -258,6 +259,64 @@ def _build_vector_query(self) -> Select:
                 DISTANCE_LABEL
             )
 
+        if self._prefilter:
+            return self._build_pre_filter_vector_query(distance_column=distance_column)
+        else:
+            return self._build_post_filter_vector_query(distance_column=distance_column)
+
+    def _build_pre_filter_vector_query(self, distance_column: Column) -> Select:
```

Comment:
Is it possible to minimize changes and avoid introducing a new function with similar code?